### PR TITLE
fix: Generate CHANGELOG.md on push to `master`

### PR DESCRIPTION
**Pull request by tjtelan**
Generates a changelog using Git log using [`git-cliff`](https://github.com/orhun/git-cliff) + a config file that I tuned to Fluvio's repo.
- We still follow [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) format
  - The Unreleased block no longer has the version number from `VERSION`, for templating purposes. The final file has this rendered.
- We are limited to commits in the Git history, and not so much Github PRs.
  - We categorize commits based on keywords in the commit message, and display commits by their subject line.
    - This means we need to keep commit titles useful.
    - Commits that meet [Conventional Commit standards](https://www.conventionalcommits.org/en/v1.0.0/) will be categorized in changelog by those standards.
- A sample (unsquashed) changelog is generated in CI and kept as a CI artifact
- A final changelog is generated and committed after push by Bors to `master`
  - The reason we make this change directly to `master` is because the commits of the PR aren't yet squashed.

Modified the release so it will still extract the latest changelog for the release notes.

Added `pr_merge.yml` workflow that triggers on push to `master`
- Changelog is generated, merged into the most recent PR commit, and force pushed to the `master` branch.
- We need a personal access token w/ `public_repo` in order to push to `master` through the branch protection.
  - `${{ secrets.CI_COMMIT_PUSH_PAT }}`

Resolves https://github.com/infinyon/fluvio/issues/1633
Closes https://github.com/infinyon/fluvio/issues/1820

### CODE REVIEW

**Comment by sehz** [comment]
Can you link to issue?

File: cliff.toml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    { pattern = "\\(#([0-9]+)\\)", replace = "([#${1}](https://github.com/infinyon/fluvio/issues/${1}))"}
+]
+
+# regex for parsing and grouping commits
+commit_parsers = [
+
+    { message = ".*[Bb]ump", group = "Noise", skip = true},
+    { message = ".*[Rr]evert", group = "Noise", skip = true},
+    { message = ".*[Cl]ippy", group = "Noise", skip = true},
+    { message = "^Merge pull request", group = "Noise", skip = true},
```
Comment (morenol) [inline cliff.toml:50]: Could you add a "[Ii]gnore.*" or "[Ss]kip.*", so we can skip a commit with that too?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -50,6 +50,40 @@ jobs:
       ##  if: github.event_name == 'push' && github.ref == 'refs/heads/staging'
       #  run: echo "TEST_RUNS='[r1,r2,r3,r4,r5]'" | tee -a $GITHUB_ENV
 
+  generate_changelog:
```
Comment (sehz) [inline .github/workflows/ci.yml:53]: Looks like this was for testing.  Is this still needed?

**Review by sehz — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (tjtelan) [reply .github/workflows/ci.yml:53]: I suppose not, but this is an early check for the changelog rendering, even if it isn't final. I don't see the harm in having it run but we can disconnect it from any requirements to pass CI?

Comment (sehz) [reply .github/workflows/ci.yml:53]: Gotcha.  The only issue is that the workflow is duplicated.  Can you use a re-usable workflow?  https://docs.github.com/en/actions/using-workflows/reusing-workflows

**Pull request closed without merging**